.github/workflows/test.yaml

```diff
@@ -23,9 +27,22 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  runs-on-labels:
+    runs-on: ['self-hosted', 'small']
+    outputs:
+      labels: ${{ steps.labels.outputs.LABELS }}
+    steps:
+      - id: labels
+        run: |
+          if ${{ inputs.self-hosted-runner }}; then
+            echo "LABELS=['self-hosted', '${{ inputs.self-hosted-runner-label }}']" >> "$GITHUB_OUTPUT"
+          else
+            echo "LABELS=['ubuntu-22.04']" >> "$GITHUB_OUTPUT"
+          fi
```

Comment:
I'm curios why this is needed? Is there some limitation of what can be put into `runs-on`? Perhaps it is that it needs to be an array?

Reply:
I'm guessing this isn't needed anymore now since the other way that Weii suggested seems to work?